### ISSUE TITLE
Visual Studio doesn't build ProjectReferences that aren't referenced when those aren't conditioned on the first appearing TargetFramework

### ISSUE THREAD

**Issue by ViktorHofer** [dotnet/project-system#8117]
Visual Studio 17.2 Preview 5
SDK: 6.0.100

Minimal repro: [p2pvs.zip](https://github.com/dotnet/runtime/files/8604129/p2pvs.zip)

Project References with the `ReferenceOutputAssembly=false` attribute on them aren't built when they are conditioned on a tfm that isn't the first one in the `TargetFrameworks` property.

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <!-- child.csproj isn't build along with p2pvs.csproj. When switching the
         order of the TargetFrameworks and reloading the projects inside VS,
         child.csproj is built as well. -->
    <TargetFrameworks>net6.0;net5.0</TargetFrameworks>
    <EnableDefaultItems>false</EnableDefaultItems>
  </PropertyGroup>
  
  <ItemGroup>
    <Compile Include="Program.cs" />
  </ItemGroup>

  <ItemGroup>
    <ProjectReference Include="child\child.csproj"
                      ReferenceOutputAssembly="false"
                      Condition="'$(TargetFramework)' == 'net5.0'" />
  </ItemGroup>

</Project>
```

### Wrong behavior
child.csproj isn't built
```
Build started...
1>------ Build started: Project: p2pvs, Configuration: Debug Any CPU ------
1>p2pvs -> C:\temp\p2pvs\bin\Debug\net6.0\p2pvs.dll
1>p2pvs -> C:\temp\p2pvs\bin\Debug\net5.0\p2pvs.dll
========== Build: 1 succeeded, 0 failed, 0 up-to-date, 0 skipped ==========
```

### Right behavior
Changing the `TargetFrameworks` string to `<TargetFrameworks>net5.0;net6.0</TargetFrameworks>` and reloading the projects makes the child.csproj project build:
```
Build started...
1>------ Build started: Project: child, Configuration: Debug Any CPU ------
1>child -> C:\temp\p2pvs\child\bin\Debug\netstandard2.0\child.dll
2>------ Build started: Project: p2pvs, Configuration: Debug Any CPU ------
2>p2pvs -> C:\temp\p2pvs\bin\Debug\net6.0\p2pvs.dll
2>p2pvs -> C:\temp\p2pvs\bin\Debug\net5.0\p2pvs.dll
========== Build: 2 succeeded, 0 failed, 0 up-to-date, 0 skipped ==========

```

cc @ericstj

**Comment by drewnoakes** [comment]
This is a known issue and belongs to CPS. We've been discussing this recently. @lifengl @adrianvmsft do you have a work item tracking this internally?

There is a possible workaround in https://github.com/dotnet/project-system/issues/6277#issuecomment-642348730.

See also:

- #7710
- #6277
- #4175
- #3758

**Comment by drewnoakes** [comment]
> It would be great if you could share the link to the internal tracking work item

https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1524343

**Comment by drewnoakes** [comment]
As the fix for this issue will be in CPS and not this repo, I am closing this.

**Comment by drewnoakes** [comment]
The CPS side of this work has completed.

**Comment by aolszowka** [comment]
@drewnoakes Just so I understand this thread here; are you saying this has been fixed in the Legacy Project System as well now? What release will this drop on?

**Comment by drewnoakes** [comment]
@aolszowka this issue was only relating to CPS (SDK-style) projects. Is there some impact on legacy projects I don't see?

It looks like the internal PR merged into 17.3, so it should be available in the latest previews if you're keen to try it out.

**Comment by aolszowka** [comment]
@drewnoakes Thank you for the clarification!
